src/main/java/ch/ivyteam/ivy/maven/AbstractIntegrationTestMojo.java

```diff
@@ -0,0 +1,62 @@
+package ch.ivyteam.ivy.maven;
+
+import java.io.File;
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+public abstract class AbstractIntegrationTestMojo extends AbstractEngineMojo
+{
+  /**
+   * Configure if the cached engine should be copied into the target folder of the maven project.
+   * This enables you to start each testing cycle with a clean engine.
+   * <ul>
+   *   <li><code>TARGET</code> = always copy the engine into the maven project target folder, this provides you with a clean engine in each test cycle.</li>
+   *   <li><code>CACHE</code> = if the engine comes from the {@link #engineCacheDirectory} copy it to the maven target folder.</li>
+   *   <li><code>IGNORE</code> = don't copy the engine, this could lead to unforeseen behaviour if the engine is used multiple times.</li>
+   * </ul>
+   */
+  @Parameter(property = "ivy.test.engine.location", defaultValue=TestEngineLocation.CACHE)
+  String testEngineLocation;
```

Comment:
@since flag missing

maybe we should tell the user why he should care about 'clean engines': what does it mean; what could be dirty?; ...

Reply:
much better now: love it